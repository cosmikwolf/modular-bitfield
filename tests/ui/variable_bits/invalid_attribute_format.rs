use modular_bitfield::prelude::*;

#[derive(Specifier)]
#[bits = hello]  // Invalid: should be an integer
enum TestData {
    #[discriminant = 0]
    Small(u8),
    #[discriminant = 1]
    Large(u16),
}

fn main() {}