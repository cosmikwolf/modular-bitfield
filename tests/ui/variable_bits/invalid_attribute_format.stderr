error: bits must be integer literal for fixed size: #[bits = N]
 --> tests/ui/variable_bits/invalid_attribute_format.rs:4:3
  |
4 | #[bits = hello]  // Invalid: should be an integer
  |   ^^^^^^^^^^^^

error: attribute value must be a literal
 --> tests/ui/variable_bits/invalid_attribute_format.rs:4:10
  |
4 | #[bits = hello]  // Invalid: should be an integer
  |          ^^^^^
